### PR TITLE
Move to uproot4

### PR DESCRIPTION
**Pull request by fcatalan92**
Some testing is still needed

### CODE REVIEW

File: ProjectDplusDsTree.py [hunk 1]
```diff
@@ -146,16 +146,18 @@
...
+        bins = ptWeights.axis(0).edges()
+        ptCentW = [(bins[iBin]+bins[iBin+1])/2 for iBin in range(len(bins)-1)]
         sPtWeights = InterpolatedUnivariateSpline(ptCentW, ptWeights.values)
         dataFramePrompt['pt_weights'] = ApplySplineFuncToColumn(dataFramePrompt, 'pt_cand', sPtWeights, 0, 50)
         if not args.ptweightsB:
             dataFrameFD['pt_weights'] = ApplySplineFuncToColumn(dataFrameFD, 'pt_cand', sPtWeights, 0, 50)
             sPtWeightsDfromB = sPtWeights
 
     if args.ptweightsB:
-        ptWeightsB = uproot.open(args.ptweights[0])[args.ptweights[1]]
```
Comment (fcatalan92) [inline ProjectDplusDsTree.py:157]: @fgrosa Could this bug affect the pp results of the paper?

Comment (fgrosa) [reply ProjectDplusDsTree.py:157]: CIao Fabio, I am not sure that I got which is the bug. Is the missing parenthesis?

Comment (fcatalan92) [reply ProjectDplusDsTree.py:157]: `args.ptweights` instead of `args.ptweightsB`

Comment (fgrosa) [reply ProjectDplusDsTree.py:157]: Right, I didn't notice it, thanks for spotting it! In pp the pT weights should be very similar between prompt and FD, so it should not make any difference, but we better check

**Review by fgrosa — APPROVED** [review]

**Comment by fgrosa** [comment]
@stefanopolitano @danielbattistini note that you need to update `hipe4ml` to `v0.0.10` (or to the `origin/dev` branch) and hence `uproot` to `uproot4` next time you pull to have a working code

**Pull request merged**